drivers/openstack/client.go

```
@@ -439,12 +444,26 @@ func (c *GenericClient) Authenticate(d *Driver) error {
 
 	provider.UserAgent.Prepend(fmt.Sprintf("docker-machine/v%s", version.VERSION))
 
+	// Configure custom TLS settings.
 	if d.Insecure {
-		// Configure custom TLS settings.
-		config := &tls.Config{InsecureSkipVerify: true}
-		transport := &http.Transport{TLSClientConfig: config}
-		provider.HTTPClient.Transport = transport
+		config = &tls.Config{InsecureSkipVerify: true}
+	} else {
+		flCa := filepath.Join(utils.GetMachineCertDir(), "ca.crt")
+		if d.TlsCACert != "" {
+			flCa = d.TlsCACert
+		}
+		certPool := x509.NewCertPool()
+		file, err := ioutil.ReadFile(flCa)
+		if err != nil {
+			log.Debug("Couldn't read ca cert %s: %s", flCa, err)
```

Here we should return the error instead of just log it.